### DISCUSSION TITLE
Dev Server port not in nitro.config.ts

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [nitrojs/nitro#2797] [Q&A]
I can't find `port` for the dev server, only the `PORT` env var works.
Is this by design or is missing that prop in `devServer` object in the config?

⚡👋

**Comment by angelhdzmultimedia** [comment]
#1 Apparently it's not loading the config from `nitro.config.ts`
#2 The port argument of server.listen is not working, only the port from the options.
![image](https://github.com/user-attachments/assets/bbaa99ef-6b22-4c20-a6d1-92eb19e6d90e)

Here I added the missing logic to load the nitro config.

Found the root of the issue #2: 
![image](https://github.com/user-attachments/assets/6074edb9-d761-40a5-8b6f-dfd9e07e4f15)
The port parameter gets overridden by the port from the options, that comes from the args and from the `PORT` env variable, which is undefined because I'm setting the port in `nuxt.config.ts` ~> `devServer.port`.
So it should be 
```ts
await listen(toNodeListener(app), { ...opts, port });`
```

**Comment by markthree** [comment]
@angelhdzmultimedia you can use [markthree/nitro-port-module](https://github.com/markthree/nitro-port-module)
